transport.go

```diff
@@ -64,10 +66,10 @@ func retryAfter(now time.Time, r *http.Response) time.Duration {
 	return defaultRetryAfter
 }
 
-func getRequestBodyFromEvent(event *Event) []byte {
+func getRequestBodyFromEvent(event *Event) ([]byte, error) {
```

Comment:
Here looks like logging was used where we could return errors.

However, most of the time when we return an error as the last argument it means the other return values are no good. Only few functions have "partial" success and that's harder to reason about.

Since this is not exported, we have more freedom to change. I'd suggest passing in the logger as an explicit argument.

```suggestion
func getRequestBodyFromEvent(event *Event, logger *log.Logger) []byte {
```